libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -194,6 +194,7 @@ void TransactionManager::saveTransactionsFromDagBlock(SharedTransactions const &
   for (uint64_t i = 0; i < trxs.size(); i++) {
     auto const &trx_hash = trx_hashes[i];
     // We only save transaction if it has not already been saved
+    markTransactionKnown(trx_hash);
```

Comment:
I don't think this is needed because these transactions should have already been marked as known. these are transactions which we are pulling from our structures and not something we received over network